CI: hypothesis occasionally failing

have seen this failure occasionally, need to see if this is an actual test case that we are not handling, or its just hypthesis flakiness. Let's fix in any event.

e.g. https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=6564

```
nclosed file <_io.TextIOWrapper name=0 mode='r' encoding='UTF-8'>
2019-01-05T14:20:09.0701278Z sys:1: ResourceWarning: unclosed <socket.socket fd=10, family=AddressFamily.AF_INET, type=SocketKind.SOCK_DGRAM, proto=0, laddr=('0.0.0.0', 0)>
2019-01-05T14:20:11.0385550Z 
2019-01-05T14:20:11.0388329Z =================================== FAILURES ===================================
2019-01-05T14:20:11.0388803Z ___________________________ test_tick_add_sub[Milli] ___________________________
2019-01-05T14:20:11.0389871Z [gw0] linux -- Python 3.7.2 /home/vsts/miniconda3/envs/pandas-dev/bin/python
2019-01-05T14:20:11.0391214Z 
2019-01-05T14:20:11.0392408Z cls = <class 'pandas.tseries.offsets.Milli'>
2019-01-05T14:20:11.0393419Z 
2019-01-05T14:20:11.0394965Z     @pytest.mark.parametrize('cls', tick_classes)
2019-01-05T14:20:11.0395812Z >   @example(n=2, m=3)
2019-01-05T14:20:11.0396155Z     @example(n=800, m=300)
2019-01-05T14:20:11.0396909Z     @example(n=1000, m=5)
2019-01-05T14:20:11.0397979Z     @given(n=st.integers(-999, 999), m=st.integers(-999, 999))
2019-01-05T14:20:11.0398357Z     def test_tick_add_sub(cls, n, m):
2019-01-05T14:20:11.0398577Z 
2019-01-05T14:20:11.0398798Z pandas/tests/tseries/offsets/test_ticks.py:41: 
2019-01-05T14:20:11.0399043Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2019-01-05T14:20:11.0399633Z ../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/hypothesis/core.py:598: in execute
2019-01-05T14:20:11.0399952Z     % (test.__name__, text_repr[0])
2019-01-05T14:20:11.0400221Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2019-01-05T14:20:11.0400423Z 
2019-01-05T14:20:11.0400647Z self = <hypothesis.core.StateForActualGivenExecution object at 0x7f75c25bfc88>
2019-01-05T14:20:11.0402646Z message = 'Hypothesis test_tick_add_sub(cls=Milli, n=-447, m=-513) produces unreliable results: Falsified on the first call but did not on a subsequent one'
2019-01-05T14:20:11.0402767Z 
2019-01-05T14:20:11.0402821Z     def __flaky(self, message):
2019-01-05T14:20:11.0402897Z         if len(self.falsifying_examples) <= 1:
2019-01-05T14:20:11.0402948Z >           raise Flaky(message)
2019-01-05T14:20:11.0403360Z E           hypothesis.errors.Flaky: Hypothesis test_tick_add_sub(cls=Milli, n=-447, m=-513) produces unreliable results: Falsified on the first call but did not on a subsequent one
2019-01-05T14:20:11.0403446Z 
2019-01-05T14:20:11.0403739Z ../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/hypothesis/core.py:803: Flaky
</details>
```

--- a/pandas/tests/tseries/offsets/test_ticks.py
+++ b/pandas/tests/tseries/offsets/test_ticks.py
@@ -4,7 +4,7 @@ Tests for offsets.Tick and subclasses
 """
 from datetime import datetime, timedelta
 
-from hypothesis import assume, example, given, strategies as st
+from hypothesis import assume, example, given, settings, strategies as st
 import numpy as np
 import pytest
 
@@ -38,6 +38,7 @@ def test_delta_to_tick():
 
 
 @pytest.mark.parametrize('cls', tick_classes)
+@settings(deadline=None)  # GH 24641
 @example(n=2, m=3)
 @example(n=800, m=300)
 @example(n=1000, m=5)

